linter/lints/crl/lints_test.go

```diff
@@ -251,3 +251,40 @@ func TestHasMozReasonCodes(t *testing.T) {
 	test.AssertEquals(t, res.Status, lint.Error)
 	test.AssertContains(t, res.Details, "MUST NOT include reasonCodes other than")
 }
+
+func TestHasValidTimestamps(t *testing.T) {
+	crl := loadPEMCRL(t, "testdata/good.pem")
+	res := hasValidTimestamps(crl)
+	test.AssertEquals(t, res.Status, lint.Pass)
+
+	// Ensure a UTCTime with '50' in the 'YY' (year) field is interpreted as
+	// '1950', not '2050'.
+	crl = loadPEMCRL(t, "testdata/good_utctime_1950.pem")
+	res = hasValidTimestamps(crl)
+	test.AssertEquals(t, res.Status, lint.Pass)
+
+	// Ensure a GeneralizedTime in the year '2050' is accepted.
+	crl = loadPEMCRL(t, "testdata/good_gentime_2050.pem")
+	res = hasValidTimestamps(crl)
+	test.AssertEquals(t, res.Status, lint.Pass)
+
+	// Check that a UTCTime with a missing 'SS' (seconds) field is rejected.
+	crl = loadPEMCRL(t, "testdata/utctime_no_seconds.pem")
+	res = hasValidTimestamps(crl)
+	test.AssertEquals(t, res.Status, lint.Error)
+	test.AssertContains(t, res.Details, "timestamps encoded using UTCTime MUST be specified in the format \"YYMMDDHHMMSSZ\"")
+
+	// Check that a GeneralizedTime with '2049' in the 'nextUpdate' field is
```

Comment:
This comment says 'nextUpdate' but the filename says 'gentime' which I'd interpret as 'ThisUpdate'. Which is it?

I like the idea of it being 'nextUpdate', just to ensure we have test cases covering all three timestamp locations.

Reply:
> This comment says 'nextUpdate' but the filename says 'gentime' which I'd interpret as 'ThisUpdate'. Which is it?

Unless I'm reading the RFC wrong, I believe all of these timestamps could be encoded in either UTCTime and GeneralizedTime.

Reply:
OHH I thought "gentime" meant "generated time" (i.e. when the notBefore timestamp is), not "generalized time". I get it now.

Reply:
I've gone ahead and rewritten all of the comments to make this much clearer.